### PR TITLE
feat: send new booking uid on reschedule webhook

### LINKED ISSUE #10631: Reschedule event webhook, payload not giving the new booking uid
### Issue Summary

So in rescheduling event the webhook which we receive it consist of payload and payload consist of uid (new booking uid) and reschedule uid (old booking uid)

Now the thing is we dont always get uid (new booking uid) in payload in case of rescheduling event. Say for example I have a slot  of 9-9:20 AM on 9 Sep I also allow 20 seats for that slot. Now say for example 2 people with different emails booked the slot on 9 Sep 9-9:20 AM. Now lets say person A rescheduled the event to 10 Sep 9-9:20 AM and when they rescheduled the event I got a webhook payload like this

```json
{
    'description': '',
    'additionalNotes': '',
    'customInputs': {},
    'startTime': '2023-09-10T03:30:00Z',
    'endTime': '2023-09-10T03:50:00Z',
    'organizer': {
        'id': 3,
        'name': 'John Sharma',
        'email': 'HIDDEN',
        'username': 'johnsharma',
        'timeZone': 'Asia/Calcutta',
        'language': {
            'locale': 'en'
        },
        'timeFormat': 'h:mma'
    },
    'responses': {
        'name': {
            'label': 'your_name',
            'value': 'Harsh Trivedi'
        },
        'email': {
            'label': 'email_address',
            'value': 'HIDDEN'
        },
        'location': {
            'label': 'location'
        },
        'title': {
            'label': 'what_is_this_meeting_about'
        },
        'notes': {
            'label': 'additional_notes'
        },
        'guests': {
            'label': 'additional_guests',
            'value': []
        },
        'rescheduleReason': {
            'label': 'reason_for_reschedule'
        },
        'itvId': {
            'label': 'Interview Id',
            'value': 'ba57f85e-bfef-48fd-be19-2b809101e8f3'
        },
        'appId': {
            'label': 'Application Id',
            'value': '4040de69-818f-4241-a2ee-3749929ba4d1'
        }
    },
    'userFieldsResponses': {
        'itvId': {
            'label': 'Interview Id',
            'value': 'ba57f85e-bfef-48fd-be19-2b809101e8f3'
        },
        'appId': {
            'label': 'Application Id',
            'value': '4040de69-818f-4241-a2ee-3749929ba4d1'
        }
    },
    'attendees': [{HIDDEN}],
    'location': '',
    'destinationCalendar': None,
    'hideCalendarNotes': False,
    'requiresConfirmation': None,
    'eventTypeId': 9,
    'seatsShowAttendees': False,
    'seatsPerTimeSlot': 20,
    'uid': '5tyzhjiZKNZubctLchWHUA',
    'attendeeSeatId': '1b0a8dfa-6908-4580-883f-416563da2cb4',
    'eventTitle': 'AI Interview',
    'eventDescription': '',
    'price': 0,
    'currency': 'usd',
    'length': 20,
    'bookingId': 47,
    'rescheduleUid': 'hk78KGqQK6Z8YX8zZQYFZe',
    'metadata': {},
    'status': 'ACCEPTED'
}

```
Notice how I got ```uid ```  variable in payload that is our new booking uid which is on 10th Sept 9-9:20AM
Now lets say user B who still had event scheduled on 9th Sept rescheduled event to 10th Sept 9-9:20AM now what has happened is the 9th Sept booking has now 0 bookings and it will get cancelled and the new booking uid is not getting created as we already have a booking on 10th Sept 9-9:20 AM so payload is not sending booking uid but this is a problem like how can we know what booking uid this is related to. The payload which I receive after User B rescheduled their booking from Sept 9  to Sept 10

```json
{
    'description': '',
    'additionalNotes': '',
    'customInputs': {},
    'startTime': '2023-09-10T03:30:00Z',
    'endTime': '2023-09-10T03:50:00Z',
    'organizer': {
        'id': 3,
        'name': 'John Sharma',
        'email': 'HIDDEN',
        'username': 'johnsharma',
        'timeZone': 'Asia/Calcutta',
        'language': {
            'locale': 'en'
        },
        'timeFormat': 'h:mma'
    },
    'responses': {
        'name': {
            'label': 'your_name',
            'value': 'Harsh Trivedi'
        },
        'email': {
            'label': 'email_address',
            'value': 'HIDDEN'
        },
        'location': {
            'label': 'location'
        },
        'title': {
            'label': 'what_is_this_meeting_about'
        },
        'notes': {
            'label': 'additional_notes'
        },
        'guests': {
            'label': 'additional_guests',
            'value': []
        },
        'rescheduleReason': {
            'label': 'reason_for_reschedule'
        },
        'itvId': {
            'label': 'Interview Id',
            'value': 'a6b10c18-ffd8-48b4-ac87-e3090a8ea97c'
        },
        'appId': {
            'label': 'Application Id',
            'value': '7d80ca85-278a-4b7b-83de-43be47dce7dd'
        }
    },
    'userFieldsResponses': {
        'itvId': {
            'label': 'Interview Id',
            'value': 'a6b10c18-ffd8-48b4-ac87-e3090a8ea97c'
        },
        'appId': {
            'label': 'Application Id',
            'value': '7d80ca85-278a-4b7b-83de-43be47dce7dd'
        }
    },
    'attendees': [{HIDDEN}],
    'location': '',
    'destinationCalendar': None,
    'hideCalendarNotes': False,
    'requiresConfirmation': None,
    'eventTypeId': 9,
    'seatsShowAttendees': False,
    'seatsPerTimeSlot': 20,
    'eventTitle': 'AI Interview',
    'eventDescription': '',
    'price': 0,
    'currency': 'usd',
    'length': 20,
    'bookingId': 50,
    'rescheduleUid': 'hk78KGqQK6Z8YX8zZQYFZe',
    'rescheduleStartTime': '2023-09-09T03:30:00Z',
    'rescheduleEndTime': '2023-09-09T03:50:00Z',
    'metadata': {},
    'status': 'ACCEPTED'
}


```

Notice here ```uid``` is missing in the payload.

### Expected Results

- In ideal condition it should have sent uid in any case whether its for booking rescheduling or creation

### Current work around
- I use the start time provided by payload to get the booking id which is accepted from Bookings table

### PR DESCRIPTION
**Pull request by wesleymatosdev**
- test: create test checking uid on `BOOKING_RESCHEDULED` webhook
- refactor: move `createUserWithSeatedEventAndAttendees` to testUtils
- refactor: add helper to create webhook receiver
- test: add test case to booking with multiple seats being rescheduled
- feat: return uid on any `BOOKING_RESCHEDULED` scenario

## What does this PR do?

Fixes #10631

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## How should this be tested?

I added e2e tests, they should be executed on the pipeline without manual interaction

## Mandatory Tasks

- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/web/playwright/webhook.e2e.ts [hunk 1]
```diff
@@ -389,6 +418,147 @@ test.describe("BOOKING_REQUESTED", async () => {
...
+      expect(webhookReceiver.requestList.length).toBe(2);
+    });
+
+    const [_, secondRequest] = webhookReceiver.requestList;
+
+    expect(secondRequest?.body).toMatchObject({
+      triggerEvent: "BOOKING_RESCHEDULED",
+      payload: {
+        // in the current implementation, it is the same as the first booking
+        uid: newBooking?.uid,
```
Comment (ologbonowiwi) [inline apps/web/playwright/webhook.e2e.ts:556]: This was `null` before the changes on this PR.

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 2]
```diff
@@ -2222,6 +2223,7 @@ async function handler(
   const webhookData = {
     ...evt,
     ...eventTypeInfo,
+    uid,
```
Comment (ologbonowiwi) [inline packages/features/bookings/lib/handleNewBooking.ts:2226]:
```suggestion
```

**Review by CarinaWolli — DISMISSED** [review]
Changes look good and tests are passing. Thank you for this fix 🙏

**Comment by CarinaWolli** [comment]
@ologbonowiwi Just saw that there are merge conflicts, could you take a look at that?

**Review by CarinaWolli — DISMISSED** [review]
fixed merge conflicts

**Review by CarinaWolli — APPROVED** [review]

**Pull request merged**